app/lib/RouteResolver.js

```diff
@@ -2,16 +2,49 @@
 
 const fs = require('fs');
 const path = require('path');
+const parse = require('url').parse;
 const _ = require('lodash');
+const pathToRegExp = require('path-to-regexp');
 const expandPath = require('../../lib/expandPath');
 const Expectation = require('./Expectation');
+
+function isEqualMethod(method1, method2) {
+  const m1 = method1.toLowerCase();
+  const m2 = method2.toLowerCase();
+  return m1 === 'all' || m2 === 'all' || m1 === m2;
+}
+
+function isRouteForRequest(route, req) {
+  if (!isEqualMethod(req.method, route.method)) return false;
+
+  const pathname = parse(req.url, true).pathname;
+
+  if (route.pathname !== '*' && !route.pathRegExp.test(pathname)) return false;
+
+  // TODO: Later add features to match other things, like query parameters, etc.
```

Comment:
This is interesting. Have you given thought to the mockyeah api, specifically what the api for matching specificity of route url params and headers?

Reply:
Yes. For query parameters, the simplest case could be something like this:
```js
mockyeah.get('/foo?bar=yes', { json: { here: 'you go' } })
```
We can parse the query parameter out of the path before registering the route, and then match on them with each request.

Or more advanced:
```js
mockyeah.get('/foo', { json: { here: 'you go' } })
  .params({
    bar: 'yes', // must be 'yes',
    color: /(red|blue)/, // must match regex
    needed: true // required, but with any value
  })
```

Any maybe even allow arrays, functions, etc., e.g.:
```js
mockyeah.get('/foo', { json: { here: 'you go' } })
  .params({
    thing: ['a', 'b'], // match 'a' or 'b'
    bar: param => param === 'yes' || param === 'no', // match 'yes' or 'no',
  })
```

Or even an overall function, to allow logic that checks complex parameter combinations, e.g.:
```js
mockyeah.get('/foo', { json: { here: 'you go' } })
  .params(params => {
    // disallow parameter "bad", and require either "say=hey&nice=yes" or "say=rawr&nice=no":
    return !params.bad && (
      (params.say === 'hey' && params.nice === 'yes')
        || (params.say === 'rawr' && params.nice === 'no')
    )
  })
```

There could be a similar method for headers, e.g.:
```js
mockyeah.get('/foo', { json: { here: 'you go' } })
  .headers({
    'X-Only-With-This': true,
    'X-My-Header': 'okay'
  })
```

And a similar method for matching request body, like `nock`, e.g. https://github.com/node-nock/nock#specifying-request-body.